services/QuillLMS/app/models/feedback_history.rb

```diff
@@ -94,11 +95,60 @@ def serializable_hash(options = nil)
     ))
   end
 
+  def serialize_by_activity_session
+   serializable_hash(only: [:session_uid, :start_date, :activity_id, :because_attempts, :but_attempts, :so_attempts, :complete], include: []).symbolize_keys
+  end
+
+  def serialize_by_activity_session_detail
+   serializable_hash(only: [:entry, :feedback_text, :feedback_type, :optimal, :used], include: []).symbolize_keys
+  end
+
   def self.batch_create(param_array)
     param_array.map { |params| create(params) }
   end
 
   private def confirm_prompt_type
     self.prompt_type = DEFAULT_PROMPT_TYPE if prompt_id && !prompt_type
   end
+
+  def self.list_by_activity_session(activity_id: nil, page: 1, page_size: DEFAULT_PAGE_SIZE)
+    query = select(%{
```

Comment:
One other request, let's use `<<~SQL` / `SQL` so we get syntax highlighting

Reply:
Good call.